File: Nadlan/ClientApp/src/app/security/app.user.auth.ts
import { AppUserClaim } from "./app.user.claim";

export class AppUserAuth {
  userName: string = "";
  bearerToken: string = "";
  isAuthenticated: boolean = false;
  claims: AppUserClaim[] = [];
  //canViewReports: boolean = false;
  //canConfirmExpenses: boolean = false;
  //canViewExpenses: boolean = false;
  //canViewTransactions: boolean = false;
  //canAccessRest: boolean = false;
}

File: Nadlan/Models/Security/AppUserAuth.cs
using System.Collections.Generic;

namespace Nadlan.Models.Security
{
    public class AppUserAuth
    {
        public AppUserAuth()
        {
            UserName = "Not authorized";
            BearerToken = string.Empty;
        }

        public string UserName { get; set; }
        public string BearerToken { get; set; }
        public bool IsAuthenticated { get; set; }
        public List<AppUserClaim> Claims { get; set; }
        //public bool CanViewReports { get; set; }
        //public bool CanViewExpenses { get; set; }
        //public bool CanConfirmExpenses { get; set; }
        //public bool CanViewTransactions { get; set; }
        //public bool CanAccessRest { get; set; }
    }
}
